demos/action_recognition_demo/python/action_recognition_demo/steps.py

```diff
@@ -26,18 +26,52 @@
 from .queue import Signal
 
 
-def run_pipeline(capture, encoder, decoder, render_fn, decoder_seq_size=16, fps=30):
+def run_pipeline(capture, model_type, model, render_fn, seq_size=16, fps=30):
     pipeline = AsyncPipeline()
     pipeline.add_step("Data", DataStep(capture), parallel=False)
-    pipeline.add_step("Encoder", EncoderStep(encoder), parallel=False)
-    pipeline.add_step("Decoder", DecoderStep(decoder, sequence_size=decoder_seq_size), parallel=False)
+
+    if model_type in ('en-de', 'dummy-de'):
+        pipeline.add_step("Encoder", EncoderStep(model[0]), parallel=False)
+        pipeline.add_step("Decoder", DecoderStep(model[1], sequence_size=seq_size), parallel=False)
+    elif model_type == 'i3d-rgb':
+        pipeline.add_step("I3DRGB", I3DRGBModelStep(model[0], seq_size, 256, 224), parallel = False)
+
     pipeline.add_step("Render", RenderStep(render_fn, fps=fps), parallel=True)
 
     pipeline.run()
     pipeline.close()
     pipeline.print_statistics()
 
 
+class I3DRGBModelStep(PipelineStep):
+
+    def __init__(self, model, sequence_size, frame_size, crop_size):
+        super().__init__()
+        self.model = model
+        assert sequence_size > 0
+        self.sequence_size = sequence_size
+        self.size = frame_size
+        self.crop_size = crop_size
+        self.input_seq = deque(maxlen = self.sequence_size)
+        self.async_model = AsyncWrapper(self.model, self.model.num_requests)
+
+    def process(self, frame):
+        preprocessed = preprocess_frame(frame, self.size, self.crop_size)
+        self.input_seq.append(preprocessed)
+        if len(self.input_seq) == self.sequence_size:
+            input_blob = np.array(self.input_seq)
+            input_blob = np.transpose(input_blob, (1, 0, 2, 3))
+            input_blob = np.expand_dims(input_blob, axis=0)
+            output, next_frame = self.async_model.infer(input_blob, frame)
```

Comment:
Doesn't this model take RGB input? Shouldn't the input be converted to RGB at some point?

Reply:
Yeah, good point, I thought, don't know why, I did it somewhere with transpose function. It's nice that you've noticed it. But, what surprise me, this model works fine on tested videos and images even with BGR channel order, without noticable changes in output  (even without changes at all, I would say). That's why I haven't noticed it myself. Thx!

Reply:
Fixed